### PR TITLE
Fix kubectl patch

### PR DESCRIPTION
**Pull request by nneram**
#32

### CODE REVIEW

File: knsk.sh [hunk 1]
```diff
@@ -201,7 +201,8 @@
           pp t4n $RES 
           if (( $DELRES )); then
             CMD1="timeout $TIME $K -n $NS --grace-period=0 --force=true delete $RES"
-            CMD2="timeout $TIME $K -n $NS patch $RES -p '{\"metadata\":{\"finalizers\":null}}'"
+            CMD2="timeout $TIME $K -n $NS patch $RES --type json \
```
Comment (thyarles) [inline knsk.sh:204]: We need to check the compatibility of this command with old kubenertes versions... do you know in which version... let me spin up some clusters and test it o check. What is your Kubernetes version? Thanks.

**Review by thyarles — COMMENTED** [review]
I'm just checking the compatibility of

File: knsk.sh (hunk 1 above)
Comment (nneram) [reply knsk.sh:204]: It work on 1.19 - 1.20 - 1.21 
Last test was on 1.19

**Comment by bengray2** [comment]
I just used this script and it worked (Rancher 2.4.8).   Only thing to note is that the --dry-run output caused the following error when copied & pasted: 

C:\kubectl> kubectl -n longhorn patch replica.longhorn.io/ap-nft-r-90d02feb --type json --patch='[ { "op": "remove", "path": "/metadata/finalizers" } ]'
error: unable to parse "'[": yaml: found unexpected end of stream

and has some bad spacing:

image


Also, thank you so much for making the script and the patch.

**Comment by thyarles** [comment]
Hi guys, let's work to make this patch happens?

**Comment by nneram** [comment]
What is missing to merge this ?

**Comment by thyarles** [comment]
Hi @nneram , you're right, the code is working fine! Thanks for you help on the fix!

**Pull request merged**